pkg/col/coldata/batch.go

```diff
@@ -80,19 +80,18 @@ func BatchSize() int {
 // MaxBatchSize is the maximum acceptable size of batches.
 const MaxBatchSize = 4096
 
-// SetBatchSizeForTests modifies batchSize variable. It should only be used in
-// tests. batch sizes greater than MaxBatchSize will return an error.
-func SetBatchSizeForTests(newBatchSize int) error {
+// SetBatchSize modifies batchSize variable. Batch sizes greater than
+// MaxBatchSize will return an error.
+func SetBatchSize(newBatchSize int) error {
 	if newBatchSize > MaxBatchSize {
 		return errors.Errorf("batch size %d greater than maximum allowed batch size %d", newBatchSize, MaxBatchSize)
 	}
 	atomic.SwapInt64(&batchSize, int64(newBatchSize))
```

Comment:
Hmm, now I see why this was only for tests. It will never be valid to change this during the run of a flow, because it's not valid to have half of a flow use batch size `n` and half of the flow use batch size `m`, like we saw with the distributed case. We would need to either keep this a testing setting, or make it so that a flow gets a batch size at the start and uses it consistently for the entire flow.

I didn't carefully read your patch before, but what this patch does doesn't make sense unfortunately. We've added the batch size variable to the flow data, but when we receive it, we set the batch size for the entire node...

I think that we need to either fix this altogether or not merge it and just fix the tests. I'm leaning toward the latter - fixing this seems like it'd take a lot of work for little gain.